src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -106,7 +115,7 @@ private static float parseTime(Element testCase) {
         return 0.0f;
     }
 
-    CaseResult(SuiteResult parent, Element testCase, String testClassName, boolean keepLongStdio) {
+    CaseResult(SuiteResult parent, Element testCase, String testClassName, boolean keepLongStdio, @CheckForNull String archiveId) {
```

Comment:
Do you need an overloaded version for the original method in case something is extending this? Maybe that's not possible or a concern.